lib/team_statistics.rb

```diff
@@ -72,4 +92,50 @@ def update_total_count(season, total_season_hash, row, team_id)
 			total_season_hash[season][1] += 1 if row[:away_goals] > row[:home_goals]
 		end
 	end
+  
+  def goals_by_team_id(team_id)
+    team_goals = [] 
+    @data[:game_teams].each do |row|
+      if row[:team_id] == team_id
+        team_goals << row[:goals].to_i
+      end
+    end
+    team_goals
+  end
+
+  def average_win_percentage_against_opponent(team_id)
```

Comment:
This is great, watching the thought process on this was very interesting. However I can see some refactors for this.